File: benchmarks/base64-decode/main.rb
require 'base64'

i1 = "SGVsbG8sIFdvcmxkIQ=="
i2 = "VGhlIHF1aWNrIGJyb3duIGZveCBqdW1wcyBvdmVyIHRoZSBsYXp5IGRvZy4="
i3 = "QWFCYkNjRGRFZUZmMTIzNDU2XyFAIyYqXigpJD0rLQ=="

before = Process.clock_gettime(Process::CLOCK_MONOTONIC, :nanosecond)

Base64.decode64(i1)
Base64.decode64(i2)
Base64.decode64(i3)

after = Process.clock_gettime(Process::CLOCK_MONOTONIC, :nanosecond)

puts after - before

File: benchmarks/base64-decode/main.go
package main

import "fmt"
import "time"
import "encoding/base64"

func main() {
	var i1 = "SGVsbG8sIFdvcmxkIQ=="
	var i2 = "VGhlIHF1aWNrIGJyb3duIGZveCBqdW1wcyBvdmVyIHRoZSBsYXp5IGRvZy4="
	var i3 = "QWFCYkNjRGRFZUZmMTIzNDU2XyFAIyYqXigpJD0rLQ=="

	var before = time.Now().UnixNano()

	_, _ = base64.StdEncoding.DecodeString(i1)
	_, _ = base64.StdEncoding.DecodeString(i2)
	_, _ = base64.StdEncoding.DecodeString(i3)

	var after = time.Now().UnixNano()

	fmt.Println(after - before)
}
